src/include/duckdb/main/profiling_info.hpp

```diff
@@ -82,7 +84,46 @@ class ProfilingInfo {
 	bool Enabled(const MetricsType setting) const;
 
 public:
+	string GetMetricAsString(MetricsType setting);
 	string GetMetricAsString(MetricsType setting) const;
 	void WriteMetricsToJSON(duckdb_yyjson::yyjson_mut_doc *doc, duckdb_yyjson::yyjson_mut_val *destination);
+
+public:
+	template <class METRIC_TYPE>
+	METRIC_TYPE &GetMetricValue(const MetricsType setting) {
+		switch (setting) {
```

Comment:
Is it possible to replace the switch with: `return metrics[setting].GetValue<METRIC_TYPE>();`?